algorithms/unit_tests/TestStdAlgorithmsAdjacentDifference.cpp

```diff
@@ -96,7 +96,7 @@ void fill_view(DestViewType dest_view, const std::string& name) {
   }
 
   else {
-    throw std::runtime_error("invalid choice");
+    Kokkos::abort("invalid choice");
```

Comment:
Here and elsewhere I think we should use googletest [`FAIL()`](https://google.github.io/googletest/reference/assertions.html#FAIL). We can also give that a message. That makes it a bit more idiomatic with googletest error reporting.

Reply:
Changed

Reply:
Had to rollback that change in a number of places where the test functions were not returning `void`

Reply:
ah, yeah i guess its really only for the top level functions then

Reply:
Actually I take back this suggestion because it seems like `FAIL()` is meant to only be used in functions that will terminate the test when they exit, such as the main test body

Reply:
Please clarify, are you asking me to revert to [33fbae2](https://github.com/kokkos/kokkos/pull/7371/commits/33fbae25a5208fc6b3d247add8bb10bb2d67225c)?

Reply:
Ok we discussed this and came to the conclusion that `FAIL()` will return, and in a function could continue test execution, but also that it probably doesn't matter.

Reply:
Will merge as is.
Here is a small experiment on the compiler explorer that confirms what Nick says above, that is it essentially behaves more like an "except" than an "assert" when fired up from a function.
https://godbolt.org/z/1a7a6o4fd